src/Adapters/Auth/apple.js

```diff
@@ -54,18 +65,44 @@ const verifyIdToken = async ({ token, id }, clientID) => {
       `auth data is invalid for this user.`
     );
   }
-  if (clientID !== undefined && jwtClaims.aud !== clientID) {
+
+  if (
+    typeof clientIDs === 'undefined' ||
+    clientIDs === undefined ||
+    clientIDs === null
+  ) {
+    throw new Parse.Error(
+      Parse.Error.OBJECT_NOT_FOUND,
+      `client ids do not exist`
+    );
+  }
+
+  if (!Array.isArray(clientIDs)) {
+    throw new Parse.Error(
+      Parse.Error.OBJECT_NOT_FOUND,
+      `clientIDs need to be an array`
+    );
+  }
+
+  if (clientIDs.length === 0) {
+    throw new Parse.Error(
+      Parse.Error.OBJECT_NOT_FOUND,
+      `need at least one client id, empty array provided`
+    );
+  }
+
+  if (!clientIDs.includes(jwtClaims.aud)) {
     throw new Parse.Error(
       Parse.Error.OBJECT_NOT_FOUND,
-      `jwt aud parameter does not include this client - is: ${jwtClaims.aud} | expected: ${clientID}`
+      `jwt aud parameter does not include this client - is: ${jwtClaims.aud} | expected: ${clientIDs}`
```

Comment:
Can you refactor this and write a test case for each error?

```
throw new Parse.Error(
   Parse.Error.OBJECT_NOT_FOUND,
  `${errorMessage}`
);
```

You can check the test coverage [here](https://codecov.io/gh/parse-community/parse-server/compare/0a65331fcbbdccf68ebee72d7dc6c8ef69067d91...83115eb54ec5bb7b65a1c80cecb1ae9a45a8f752/diff) and improve it.

Reply:
So I tried to write a test case for each, following the GitHub page errors: https://github.com/auth0/node-jsonwebtoken. But since we are using apple tokens, the tokens have to be real, as they are verified with the apple public key so I keep getting a invalid signature error & using real tokens would not be ideal even if generated through a dummy app due to the identifier & email in them.

Reply:
If you have any suggestions to tackle this please let me know